internal/provider/resource_user_alias.go

```diff
@@ -0,0 +1,243 @@
+
+	}, bOff)
+
+	d.SetId(fmt.Sprintf("%s/%s", primaryEmail, alias.Alias))
+	d.Set("primary_email", primaryEmail)
+	d.Set("alias", alias.Alias)
+	return resourceUserAliasRead(ctx, d, meta)
+}
+
+func resourceUserAliasRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var diags diag.Diagnostics
+
+	client := meta.(*apiClient)
+
+	directoryService, diags := client.NewDirectoryService()
+	if diags.HasError() {
+		return diags
+	}
+
+	usersService, diags := GetUsersService(directoryService)
+	if diags.HasError() {
+		return diags
+	}
+
+	aliasesService, diags := GetUserAliasService(usersService)
+	if diags.HasError() {
+		return diags
+	}
+
+	primaryEmail := d.Get("primary_email").(string)
+	expectedAlias := d.Get("alias").(string)
+
+	resp, err := aliasesService.List(primaryEmail).Do()
+	if err != nil {
+		return diag.Errorf("[ERROR] could not retrieve aliases for user (%s): %v", primaryEmail, err)
+	}
+
+	alias, ok := doesAliasExist(resp, expectedAlias)
+	if !ok {
+		log.Println(fmt.Sprintf("[WARN] no matching alias (%s) found for user (%s).", expectedAlias, primaryEmail))
+		d.SetId("")
+		return nil
+	}
+	d.SetId(fmt.Sprintf("%s/%s", alias.PrimaryEmail, alias.Alias))
+	d.Set("primary_email", alias.PrimaryEmail)
+	d.Set("alias", alias.Alias)
+	d.Set("etag", alias.Etag)
+	return nil
+}
+
+func resourceUserAliasDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var diags diag.Diagnostics
+
+	client := meta.(*apiClient)
+
+	directoryService, diags := client.NewDirectoryService()
+	if diags.HasError() {
+		return diags
+	}
+
+	usersService, diags := GetUsersService(directoryService)
+	if diags.HasError() {
+		return diags
+	}
+
+	aliasesService, diags := GetUserAliasService(usersService)
+	if diags.HasError() {
+		return diags
+	}
+
+	primaryEmail := d.Get("primary_email").(string)
+	alias := d.Get("alias").(string)
+
+	err := aliasesService.Delete(primaryEmail, alias).Do()
+	if err != nil {
+		return diag.Errorf("[ERROR] unable to remove alias (%s) from user (%s): %v", alias, primaryEmail, err)
+	}
+
+	d.SetId("")
+	return nil
+}
+
+func resourceUserAliasImport(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	var diags diag.Diagnostics
+
+	client := meta.(*apiClient)
+
+	directoryService, diags := client.NewDirectoryService()
+	if diags.HasError() {
+		return nil, fmt.Errorf("[ERROR] Unable to init client: %s", diags[0].Summary)
+	}
+
+	usersService, diags := GetUsersService(directoryService)
+	if diags.HasError() {
+		return nil, fmt.Errorf("[ERROR] Unable to init client: %s", diags[0].Summary)
+	}
+
+	aliasesService, diags := GetUserAliasService(usersService)
+	if diags.HasError() {
+		return nil, fmt.Errorf("[ERROR] Unable to init client: %s", diags[0].Summary)
+	}
+
+	primaryEmail := strings.Split(d.Id(), "/")[0]
+	expectedAlias := strings.Split(d.Id(), "/")[1]
+
+	resp, err := aliasesService.List(primaryEmail).Do()
```

Comment:
You shouldn't have to make the API call here. After import, the Read function is called, so you just need to make sure that `primary_email` and `alias` have been set. See `googleworkspace_group_member` for an [example](https://github.com/hashicorp/terraform-provider-googleworkspace/blob/main/internal/provider/resource_group_member.go#L275).